File: bindings/python/examples/provider_embed.py
from __future__ import annotations

from _common import (
    load_openai_provider_client,
    print_embedding,
    provider_endpoint,
    read_provider_args,
)


def main() -> None:
    model, input_text = read_provider_args("CogentClient provider embedding smoke input.")
    client = load_openai_provider_client(model)
    run = client.embed(
        input_text,
        endpoint=provider_endpoint(model),
    )
    print_embedding(run.result())


if __name__ == "__main__":
    main()

File: bindings/node/examples/provider_embed.mjs
import {
  loadOpenAiProviderClient,
  printEmbedding,
  providerEndpoint,
  readProviderArgs,
} from './_common.mjs';

const { model, input } = readProviderArgs('CogentClient provider embedding smoke input.');
const client = loadOpenAiProviderClient(model);
const result = await client.embed({
  endpoint: providerEndpoint(model),
  input,
}).response;
printEmbedding(result);
